pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -462,6 +483,7 @@ func (p *PriorityQueue) Update(oldPod, newPod *v1.Pod) error {
 func (p *PriorityQueue) Delete(pod *v1.Pod) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+	delete(p.processingPods, pod.UID)
```

Comment:
What about deleting `processingPods` on Pod Update?

Reply:
that's a bit aggressive... I'd suggest to do it in `Delete()` for now.

Reply:
sg. I saw it as a way to skip checking for pod in active queue when doing `moveToUnschedulable` . But yeah, it's pretty much equivalent and what you have may have better semantics.